MobileCenter/MobileCenter/Internals/Channel/MSLogManagerDelegate.h

```diff
@@ -1,46 +1,78 @@
- *  also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
- *  method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
- *  log.
+ * @discussion We had some discussion about the naming of the method. To match the
+ * onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
+ * also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
+ * method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
+ * log.
  */
 - (void)onFinishedPersistingLog:(id<MSLog>)log withInternalId:(NSString *)internalId;
 
 /**
- *  Callback that is called when persisting a log has failed, meaning it has not been saved to disk because the log was
- *  empty. This was introduced to implement the log buffer for Crashes.
+ * Callback that is called when persisting a log has failed, meaning it has not been saved to disk because the log was
+ * empty. This was introduced to implement the log buffer for Crashes.
  *
- *  @param log The log.
- *  @param internalId An internal Id that can be used to keep track of logs.
+ * @param log The log.
+ * @param internalId An internal Id that can be used to keep track of logs.
  *
- *  @discussion We had some discussion about the naming of the method. To match the
- *  onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
- *  also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
- *  method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
- *  log.
+ * @discussion We had some discussion about the naming of the method. To match the
+ * onEnqueueingLog:withInternalId (@see onEnqueueingLog:withInternalId) callback, it should
+ * also use `enqueuing` in it's signature, yet, as of now, it indicates successful persistence of a log. This
+ * method's name might change in the future to make a distinction between a successfully enqueued log and a persisted
+ * log.
  */
 - (void)onFailedPersistingLog:(id<MSLog>)log withInternalId:(NSString *)internalId;
 
+/**
+ * Callback method that will be called before each log will be send to the server.
+ *
+ * @param log The log to be sent.
+ */
+- (void)willSendLog:(id<MSLog>)log;
+
+/**
+ * Callback method that will be called in case the SDK was able to send a log.
+ *
+ * @param log The log to be sent.
+ */
+- (void)didSucceedSendingLog:(id<MSLog>)log;
+
+/**
+ * Callback method that will be called in case the SDK was unable to send a log.
+ *
+ * @param log The log to be sent.
+ * @param error The error that occured.
+ */
+- (void)didFailSendingLog:(id<MSLog>)log withError:(NSError *)error;
+
+/**
+ * Get service unique key for storage purpose.
```

Comment:
Because this class is about delegates not services so it feels weird to have service mentioned here. For instance when we refer to group Id in log manager or a channels we don't mention service because it's abstracted, we don't need to know who is using it, we just need to know the purpose of it.